Marlin/Conditionals_post.h

```diff
@@ -731,7 +731,7 @@
    * Set granular options based on the specific type of leveling
    */
 
-  #define UBL_DELTA  (ENABLED(AUTO_BED_LEVELING_UBL) && ENABLED(DELTA))
+  #define UBL_DELTA  (ENABLED(AUTO_BED_LEVELING_UBL) && (ENABLED(DELTA) || ENABLED(UBL_GRANULAR_SEGMENTATION_FOR_CARTESIAN)))
```

Comment:
So… extending our repertoire of general flags with this…
```cpp
#define MOVE_SEGMENTATION (IS_KINEMATIC || ENABLED(CARTESIAN_SEGMENTATION))
```
…We will then have a simple conditional that we can use to decide whether to include and to call the `prepare_kinematic_move_to` (rename to `prepare_segmented_move_to`?) function which splits up moves for a kinematic system, or whether to call the normal movement stuff — which should still preserve the split-on-mesh-boundaries code.